src/components/modals/ModalWithActionButtons.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Button, ButtonProps } from '../buttons/Button';
+
+import { BaseModal } from './BaseModal';
+
+export interface ActionButton extends ButtonProps {
+  actionCallback?: (...params: any[]) => any;
+  closeModalOnClick?: boolean;
+}
+
+export interface ModalProps {
+  title: string;
+  contentText: string;
+  open: boolean;
+  handleClose: () => void;
+}
+
+interface ModalActionButtonProps {
+  modalProps: ModalProps;
+  actionButtonList: ActionButton[];
+}
+
+export const ModalWithActionButtons = ({
+  modalProps,
+  actionButtonList,
+}: ModalActionButtonProps) => {
+  return (
+    <BaseModal {...modalProps}>
+      {actionButtonList.map((buttonProps: ActionButton) => {
+        const {
+          name,
+          closeModalOnClick = true,
+          actionCallback = () => {
+            return null;
+          },
+          ...otherProps
+        } = buttonProps;
+
+        const onClickFunction = () => {
+          if (closeModalOnClick) {
```

Comment:
maybe call action call back then close, later on we can throw up a loading as well 
otherwise ppl might close then do other stuff

pure speculation for this though